javaslang-benchmark/src/test/java/javaslang/MemoryUsage.java

```diff
@@ -0,0 +1,36 @@
+package javaslang;
```

Comment:
yes, it does, this is the memory consumption of the new development version of `Vector`:
```

for 32 elements
	`java.util.ArrayList` uses `504 bytes` (`0 bytes` overhead, `0.0` bytes overhead per element)
	`scala.collection.immutable.Vector` uses `536 bytes` (`32 bytes` overhead, `1.0` bytes overhead per element)
	`clojure.lang.PersistentVector` uses `704 bytes` (`200 bytes` overhead, `6.2` bytes overhead per element)
	`org.pcollections.TreePVector` uses `1.7 KB` (`1.2 KB` overhead, `38.8` bytes overhead per element)
	`fj.data.Seq` uses `3 KB` (`2.5 KB` overhead, `80.0` bytes overhead per element)
	`javaslang.collection.Vector` uses `536 bytes` (`32 bytes` overhead, `1.0` bytes overhead per element)

for 1024 elements
	`java.util.ArrayList` uses `18.6 KB` (`0 bytes` overhead, `0.0` bytes overhead per element)
	`scala.collection.immutable.Vector` uses `19.3 KB` (`672 bytes` overhead, `0.7` bytes overhead per element)
	`clojure.lang.PersistentVector` uses `20 KB` (`1.4 KB` overhead, `1.4` bytes overhead per element)
	`org.pcollections.TreePVector` uses `54.7 KB` (`36.1 KB` overhead, `36.1` bytes overhead per element)
	`fj.data.Seq` uses `102.3 KB` (`83.7 KB` overhead, `83.7` bytes overhead per element)
	`javaslang.collection.Vector` uses `20 KB` (`1.4 KB` overhead, `1.4` bytes overhead per element)

for 32768 elements
	`java.util.ArrayList` uses `638.4 KB` (`0 bytes` overhead, `0.0` bytes overhead per element)
	`scala.collection.immutable.Vector` uses `659 KB` (`20.7 KB` overhead, `0.6` bytes overhead per element)
	`clojure.lang.PersistentVector` uses `683.8 KB` (`45.4 KB` overhead, `1.4` bytes overhead per element)
	`org.pcollections.TreePVector` uses `1.7 MB` (`1.1 MB` overhead, `36.0` bytes overhead per element)
	`fj.data.Seq` uses `3.3 MB` (`2.6 MB` overhead, `84.1` bytes overhead per element)
	`javaslang.collection.Vector` uses `683.8 KB` (`45.4 KB` overhead, `1.4` bytes overhead per element)
```
i.e. currently it has `~30x` less memory overhead (and is around `5x` faster).
Still not finished, though.